Write correct number of ciphersuites in log

RonEld opened the pull request:
## Description
Change location of log, to fit the correct number of used ciphersuites
Fixes #918

## Status
**READY**

## Requires Backporting
When there is a bug fix, it should be backported to all maintained and supported branches.
Changes do not have to be backported if:
- This PR is a new feature\enhancement
- This PR contains changes in the API. If this is true, and there is a need for the fix to be backported, the fix should be handled differently in the legacy branch

Yes | NO  
mbedtls-1.3
mbedtls-2.1

## Migrations
NO

## Todos
- [ ] Tests
- [ ] Documentation
- [x] Changelog updated
- [ ] Backported


## Steps to test or reproduce
use a small amount of ciphersuites, enable SSL_RENEGOTIATION and\or SSL_FALLBACK_SCSV and look at log

andresag01 commented on library/ssl_cli.c:
```diff
@@ -891,6 +891,8 @@ static int ssl_write_client_hello( mbedtls_ssl_context *ssl )
         *p++ = (unsigned char)( ciphersuites[i]      );
     }
 
+	MBEDTLS_SSL_DEBUG_MSG(3, ("client hello, got %d ciphersuites", n));
```
Style: Please add space after and before `(` and `)` respectively.

andresag01 commented on library/ssl_cli.c:
```diff
@@ -891,6 +891,8 @@ static int ssl_write_client_hello( mbedtls_ssl_context *ssl )
     }
 
+	MBEDTLS_SSL_DEBUG_MSG(3, ("client hello, got %d ciphersuites", n));
+
```
Perhaps we should add a log message  when adding `MBEDTLS_SSL_EMPTY_RENEGOTIATION_INFO`?

RonEld commented:
Hi @andresag01 , I updated according to your comments

hanno-arm commented on ChangeLog:
```diff
@@ -13,6 +13,7 @@ Bugfix
      Found by redplait #590
    * Add MBEDTLS_MPI_CHK to check for error value of mbedtls_mpi_fill_random.
      Reported and fix suggested by guidovranken in #740
+   * Log correct number of ciphersuites used in Client Hello message. Fix for #918.
```
After @andresag01 suggestions, I'd prefer a more specific entry here like "Modify debug output of ssl_write_client_hello to not consider SCSV entries in the ciphersuite count but to log them separately.".

andresag01 replied:
I personally do not mind the ChangeLog entry. Actually, I think that the change is so small that it probably should not have a ChangeLog entry...

hanno-arm replied:
What about keeping it but moving it to the Changes section in it's extended form?

RonEld replied:
I agree with Andres, that this is a minor change, only on a log message, which doesn't require a ChangeLog Entry

andresag01 commented on library/ssl_cli.c:
```diff
@@ -891,13 +891,16 @@ static int ssl_write_client_hello( mbedtls_ssl_context *ssl )
         *p++ = (unsigned char)( ciphersuites[i]      );
     }
 
+	MBEDTLS_SSL_DEBUG_MSG( 3, ( "client hello, got %d ciphersuites (excluding SCSVs)", n ) );
```
Style: Please use spaces instead of tabs

andresag01 commented on library/ssl_cli.c:
```diff
@@ -891,13 +891,16 @@ static int ssl_write_client_hello( mbedtls_ssl_context *ssl )
     if( ssl->renego_status == MBEDTLS_SSL_INITIAL_HANDSHAKE )
 #endif
     {
+		MBEDTLS_SSL_DEBUG_MSG( 3, ( "adding EMPTY_RENEGOTIATION_INFO_SCSV" ) );
```
Style: Please use spaces instead of tabs

RonEld commented:
Hi @andresag01 @hanno-arm I addressed your comments. Although I agree with Andres that this PR doesn''t necessary need a ChangeLog entry, I modified the ChangeLog according to Hanno's comments, as I also saw duplication in the BugFix section

andresag01 commented:
@RonEld: There seems to be a few conflicts with the ChangeLog. You should not have to delete duplicated entries. Could you please rebase your changes on top of the current development. I know this affects a little the review history, but I think it would be good in this case. Thanks!

RonEld commented:
@andresag01 I rebased and fixed conflicts.
Please approve

gilles-peskine-arm commented:
This fixes a debug message, so it's low-risk. Can you please make a 2.1 backport? This PR is old enough to be mergeable into 2.7 so we don't need a separate PR for 2.7.
